erpnext/stock/get_item_details.py

```diff
@@ -1353,9 +1352,12 @@ def get_valuation_rate(item_code, company, warehouse=None):
 		).run()
 
 		if valuation_rate:
-			return {"valuation_rate": valuation_rate[0][0] or 0.0}
-	else:
-		return {"valuation_rate": 0.0}
+			valuation_rate = valuation_rate[0][0] or 0.0
+
+	if not valuation_rate:
+		valuation_rate = item.get("valuation_rate", 0.0)
```

Comment:
This get's set in two unrealistic scenarios:
- When warehouse is group (or parent). This should never be allowed and maybe it's validated at a different place.
- When item is not a stock item (although field is hidden, it may still carry incorrect value set before)

Reply:
Any value is better than zero in both situations.

And I don't understand why the value is hidden for non-stock items, everything for sale should have a value in most cases.

Reply:
Plus I don't think it's possible to change a stock item with a value back to a non-stock item

Reply:
It's possible if there are no transactions against item.

> Any value is better than zero in both situations.
Only if the value is actually required.

Reply:
If there are no transactions then the item is not in use so there is unlikely to be a valuation rate anyway